internal/cluster/finalizing_stages_test.go

```diff
@@ -0,0 +1,76 @@
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+)
+
+var _ = DescribeTable("finalizing stage timeouts",
```

Comment:
I can't understand what any of this is testing.

Maybe don't use `DescribeTable` here.

I'd prefer something readable and repetitive than something that is impossible to read.

Reply:
Generation of test cases by code will be always unclear whatever way to run them we choose.

Reply:
Refactored the code.  Hope it is clearer now.

Reply:
This is definitely an improvement, thanks.

I have one more suggestion, but I'll put it in a separate comment and resolve this one.